src/app/app.component.spec.ts

```diff
@@ -1,35 +1,26 @@
-import { TestBed, async } from '@angular/core/testing';
-import { RouterTestingModule } from '@angular/router/testing';
-import { AppComponent } from './app.component';
+import { TestBed, async } from "@angular/core/testing";
+import { RouterTestingModule } from "@angular/router/testing";
+import { AppComponent } from "./app.component";
+import { CUSTOM_ELEMENTS_SCHEMA } from "@angular/core";
 
-describe('AppComponent', () => {
-  beforeEach(async(() => {
-    TestBed.configureTestingModule({
-      imports: [
-        RouterTestingModule
-      ],
-      declarations: [
-        AppComponent
-      ],
-    }).compileComponents();
-  }));
+describe("AppComponent", () => {
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [RouterTestingModule],
+            declarations: [AppComponent],
+            schemas: [CUSTOM_ELEMENTS_SCHEMA]
+        }).compileComponents();
+    }));
 
-  it('should create the app', () => {
-    const fixture = TestBed.createComponent(AppComponent);
-    const app = fixture.debugElement.componentInstance;
-    expect(app).toBeTruthy();
-  });
+    it("should create the app", () => {
+        const fixture = TestBed.createComponent(AppComponent);
```

Comment:
you can assign fixture in before each
so, it should be another one, not async before each, or, you can use .then() with compileComponents cause compileComponents returns promise

Reply:
Try to move into beforeEach every repeating action in your test.
If some action is needed for some of the tests, you can use nested describes to group your unit tests